collector/pg_connections_by_client.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2024 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collector
+
+import (
+	"context"
+	"database/sql"
+	"log/slog"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const connectionsByClientSubsystem = "connections_by_client"
+
+func init() {
+	registerCollector(connectionsByClientSubsystem, defaultEnabled, NewPGConnectionsByClientCollector)
+}
+
+type PGConnectionByClientCollector struct {
+	log *slog.Logger
+}
+
+func NewPGConnectionsByClientCollector(config collectorConfig) (Collector, error) {
+	return &PGConnectionByClientCollector{
+		log: config.logger,
+	}, nil
+}
+
+var (
+	pgClientCountDesc = prometheus.NewDesc(
+		prometheus.BuildFQName(
+			namespace,
+			connectionsByClientSubsystem,
+			"count",
+		),
+		"Number of clients",
+		[]string{"name"}, nil,
+	)
+
+	pgConnectionsByClientQuery = `
+	SELECT
+		count(*) as count,
+		client_hostname
```

Comment:
This seems like a bad field to use for this data. According to the docs, this is the reverse dns of the IP. I would prefer to see this use the IP. 

https://www.postgresql.org/docs/17/monitoring-stats.html#MONITORING-PG-STAT-ACTIVITY-VIEW

Reply:
I agree, I would prefer this to be `client_addr`.

```suggestion
		client_addr
```